### PR TITLE
424 add cdk for deployment

### PR DESCRIPTION
**Pull request by JackVCurtis**
This PR:

1. Adds the AWS CDK 
2. Adds CDK code defining our AWS Infrastructure
3. Adds a Github Action that can be manually triggered to deploy infrastructure
4. Adds instructions for generating the AWS Certificate and the role to allow GitHub Actions permission to use AWS resources.

EDIT: I've also included the updated Github Action for deploying the app itself, and added some steps to the README for finalizing the DNS update

### CODE REVIEW

**Review by ngmaloney — APPROVED** [review]
I am not all that familiar with CDK so wouldn't be able to get into the weeds on functionality here. At a high-level this makes sense and seems comprehensive. If it works, I'd say :shipit:

File: AWS_SETUP.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+```bash
+npm i
+npm run bootstrap
+```
+
+Create secret values in AWS encrypted SSM Parameter store:
+
+```bash
+export DPDASH_MONGODB_ADMIN_USER=dpdash
+export DPDASH_IMPORT_API_USERS=user1,user2
```
Comment (mikestone14) [inline AWS_SETUP.md]: what is an import api user? Is this supposed to be a property of a user document in the environment we're deploying to, like the ID?

**Review by mikestone14 — COMMENTED** [review]
This looks great, just 1 clarifying question.

File: AWS_SETUP.md (hunk 1 above)
Comment (JackVCurtis) [reply AWS_SETUP.md]: Yes, it's an ID for everyone who is using the import API endpoint to add study data.

**Pull request merged**